src/map/script.cpp

```diff
@@ -25686,6 +25722,281 @@ BUILDIN_FUNC( openstylist ){
 #endif
 }
 
+BUILDIN_FUNC(getitemdata)
+{
+	struct map_session_data* sd;
+
+	if (!script_rid2sd(sd)) {
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int idx = script_getnum(st, 2);
+	if (idx < 0 || idx >= MAX_INVENTORY) {
+		ShowError("buildin_getitemdata: Index %d is out of the range 0-%d.\n", idx, MAX_INVENTORY - 1);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (sd->inventory_data[idx] == nullptr) {
+		ShowError("buildin_getitemdata: No item found on index %d of player %s (AID: %u, CID: %u).\n", idx, sd->status.name, sd->status.account_id, sd->status.char_id);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+	if (!itemdb_isequip2(sd->inventory_data[idx]) || sd->inventory_data[idx]->type == IT_AMMO) {
+		ShowError("buildin_getitemdata: Invalid type for item %u. The item must be equippable.\n", sd->inventory_data[idx]->nameid);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
```

Comment:
Just a thought for the "set" function:
Should we block calls if the target item is equipped or in equip switch?
And/or should we run status calculations after changing data?